### ISSUE TITLE
set_trace() only works the first time

### ISSUE THREAD

**Issue by killua8p** [romanvm/python-web-pdb#18]
The breakpoint (as in `import web_pdb; web_pdb.set_trace()`) works fine for the first time.
But once the debugger session is ended (e.g. by typing `c` and pressing `enter`), it won't stop again until the dev server is restarted. By "won't stop" I mean the breakpoint no longer works.

I'm using Django==1.11.22 and web-pdb==1.5.1 in a Docker container. The breakpoint was inserted in the beginning of a view function.

```
def view_func(request):
    import web_pdb; web_pdb.set_trace()
    ...
```

**Comment by romanvm** [comment]
Do you mean that calling the same view again no longer triggers a breakpoint?

**Comment by killua8p** [comment]
Ah, yes. Calling the same view again no longer triggers a breakpoint. 
I also tried using pdb at the same time to rule out caching issues. When the view was called the second time, only the pdb breakpoint was triggered.
```
def view_func(request):
    import pdb; pdb.set_trace()
    import web_pdb; web_pdb.set_trace()
    ...
```

**Comment by romanvm** [comment]
Web-PDB was never tested inside Docker so anything may happen. What Docker/Django log says when you are trying to stop on `set_trace()` for the second time?

**Comment by killua8p** [comment]
Log doesn't say anything about breakpoint, even for the first time when it works. Is it supposed to?

**Comment by maiamcc** [comment]
👍 also seeing this behavior :( any news on this ticket?

**Comment by romanvm** [comment]
As I said I never tested Web-PDB inside Docker, so definite steps to reproduce the issue would be a good start.

**Comment by jadkik** [comment]
It seems to be working for me. I made a minimal example in [this gist](https://gist.github.com/jadkik/af3d2dc9184d6598d8f9a0faecc8d50a) using `aiohttp`.

If you type `c` at the prompt, it continues normally and if you reload the debugger page you see the last "state" you were in. The next time it hits a breakpoint it also behaves as expected.

If you type `q` at the prompt, you get a `BdbQuit` exception and if you reload the debugger page it seems to be down. However the next time it hits a breakpoint (i.e. another HTTP call), you reload the debugger page and it works as expected.

I am not sure if the observed behaviour in this issue is related to `aiohttp` (async) vs `django` (non-async), rather than Docker-specific things.

You should see in the logs (`Web-PDB: starting web-server on container-id:5555`) and (`Web-PDB: stopping web-server...` then `Web-PDB: web-server stopped.`)

Hope this helps in narrowing down the issue.

**Comment by maiamcc** [comment]
Hey folks, I spent this afternoon poking into this, and I think it is *not* a container issue, but rather a webpdb-and-threads issue. I was able to reproduce the issue locally with [this toy flask app](https://gist.github.com/maiamcc/8081319294438e6bc85c3d7868f914ed); as far as I can tell, when webpdb exits due to a thread closing (which in most Python webserver frameworks is going to happen at the end of any web request), it leaves out a bit of cleanup; whenever the `set_trace()` command is hit subsequently, webpdb thinks that it's still running (because of the neglected cleanup step from before) and so doesn't refresh the web console connection, so the debugger doesn't work.

Opening a PR now.
